Sources/NetShears/URLProtocol/NetworkIgnore.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  NetworkIgnore.swift
+//  
+//
+//  Created by Mehrdad Goodarzi(Arash) on 6/24/22.
+//
+
+import Foundation
+
+public enum Ignore {
+    case disbaled
+    case enabled(ignoreHandler: ((URLRequest) -> Bool)?, grpcIgnoreHandler: ((NetShearsRequestModel) -> Bool)?, onFeatures: [IgnoringFeature])
+
+    // ignore for REST calls
+    static func shouldIgnore(request: URLRequest, on feature: IgnoringFeature) -> Bool {
+        guard case let Ignore.enabled(ignoreHandler,_,ignoringFeatures) = NetShears.shared.ignore else {
+            return false
+        }
+        guard let handler = ignoreHandler else {
+            return false
+        }
+        guard ignoringFeatures.contains(feature) else {
+            return false
+        }
+        return handler(request)
+    }
+
+    // ignore for GRPC calls
+    static func shouldIgnore(netShearsRequest: NetShearsRequestModel, on feature: IgnoringFeature) -> Bool {
+        guard case let Ignore.enabled(_,grpcIgnoreHandler,ignoringFeatures) = NetShears.shared.ignore else {
+            return false
+        }
+        guard let handler = grpcIgnoreHandler else {
+            return false
+        }
+        guard ignoringFeatures.contains(feature) else {
+            return false
+        }
+        return handler(netShearsRequest)
+    }
+}
+
+public enum IgnoringFeature {
+    case interceptor
+    case logger
+    case listener
+}
+
+/*
+func example() {
```

Comment:
Can you remove this example and write some unit tests for it

Reply:
approach changed. please resolve outdated comments